packages/sdk/src/catalog/CatalogClient.ts

```diff
@@ -1 +1,371 @@
+      });
+
+      const data = await this._client.get<{ sourceFile: FileContent }>(
+        queryString,
+      );
+      return Promise.resolve(data.sourceFile);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async processCatalogFiles(
+    props: ProcessCatalogFilesRequest,
+  ): Promise<ProcessCatalogFilesResponse> {
+    const { fileUids, namespaceUid } = props;
+
+    if (!namespaceUid) {
+      return Promise.reject(new Error("namespaceUid not provided"));
+    }
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/catalogs/files/processAsync`,
+      });
+
+      const additionalHeaders = getInstillAdditionalHeaders({
+        requesterUid: namespaceUid,
+      });
+
+      const data = await this._client.post<{ files: File[] }>(queryString, {
+        body: JSON.stringify({ file_uids: fileUids }),
+        additionalHeaders,
+      });
+
+      return Promise.resolve(data.files);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listChunks(
+    props: ListChunksRequest & { enablePagination: true },
+  ): Promise<ListChunksResponse>;
+  async listChunks(
+    props: ListChunksRequest & { enablePagination: false },
+  ): Promise<Chunk[]>;
+  async listChunks(
+    props: ListChunksRequest & { enablePagination: boolean },
+  ): Promise<ListChunksResponse | Chunk[]> {
+    const {
+      namespaceId,
+      catalogId,
+      fileUid,
+      pageSize,
+      pageToken,
+      enablePagination,
+    } = props;
+
+    try {
+      const baseUrl = `/namespaces/${namespaceId}/catalogs/${catalogId}/chunks?fileUid=${fileUid}`;
+
+      const queryString = getQueryString({
+        baseURL: baseUrl,
+        pageSize,
+        pageToken,
+      });
+
+      const data = await this._client.get<ListChunksResponse>(queryString);
+
+      if (enablePagination) {
+        return Promise.resolve(data);
+      }
+
+      const chunks = data.chunks || [];
+      if (data.nextPageToken) {
+        chunks.push(
+          ...(await this.listChunks({
+            namespaceId,
+            catalogId,
+            fileUid,
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination: false,
+          })),
+        );
+      }
+
+      return Promise.resolve(chunks);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async getChunkContent(
+    props: GetChunkContentRequest,
+  ): Promise<GetChunkContentResponse> {
+    const { namespaceId, catalogId, chunkUid } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/catalogs/${catalogId}/chunks/${chunkUid}/content`,
+      });
+
+      const data = await this._client.get<{ content: ChunkContent }>(
+        queryString,
+      );
+      return Promise.resolve(data.content);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
```

Comment:
@thewbuk Do we have this one on the docs? I can't find it

Reply:
I think we had it ages ago. Removed it